samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/AgentServices.cs

```diff
@@ -0,0 +1,433 @@
+                            Description = description.Description,
+                            Prompt = description.Prompt,
+                            ImageUrl = imageUrl,
+                            Caption = description.Caption
+                        });
+                        
+                        Logger.LogInformation($"Successfully generated image: {description.Description}");
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, $"Error generating image for prompt: {description.Prompt}");
+                        AddPlaceholderImage(generatedImages, description);
+                    }
+                }
+            }
+            else
+            {
+                // DALL-E is not configured, use placeholders instead
+                Logger.LogWarning("DALL-E is not configured. Using placeholder images instead.");
+                
+                foreach (var description in imageDescriptions)
+                {
+                    AddPlaceholderImage(generatedImages, description);
+                }
+            }
+            
+            // Return the generated images as JSON
+            return System.Text.Json.JsonSerializer.Serialize(generatedImages);
+        }
+        catch (Exception ex)
+        {
+            Logger.LogError(ex, "Error processing image descriptions");
+            return "[]";
+        }
+    }
+    
+    /// <summary>
+    /// Adds a placeholder image to the list of generated images
+    /// </summary>
+    private void AddPlaceholderImage(List<GeneratedImage> images, ImageDescription description)
+    {
+        images.Add(new GeneratedImage
+        {
+            Description = description.Description,
+            Prompt = description.Prompt,
+            ImageUrl = "https://via.placeholder.com/800x600?text=Image+Generation+Sample",
+            Caption = description.Caption
+        });
+        
+        Logger.LogInformation($"Added placeholder image for: {description.Description}");
+    }
+    
+    /// <summary>
+    /// Calls the DALL-E API to generate an image from a prompt using Microsoft Entra ID authentication
+    /// </summary>
+    /// <param name="prompt">The image generation prompt</param>
+    /// <returns>URL of the generated image</returns>
+    private async Task<string> GenerateImageWithDallE(string prompt)
+    {
+        if (!_dalleEnabled || _httpClient == null || _dallEEndpoint == null || _credential == null)
+        {
+            throw new InvalidOperationException("DALL-E is not enabled. Check environment variable DALLE_ENDPOINT.");
+        }
+        
+        // Create the request body for DALL-E API
+        var requestBody = new
+        {
+            prompt = prompt,
+            n = 1,
+            size = "1024x1024",
+            response_format = "url",
+            quality = "standard"
+        };
+        
+        var content = new StringContent(
```

Comment:
`StringContent` implements `IDisposable` and therefore needs to be disposed when you're done with it.
```suggestion
        using var content = new StringContent(
```